CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+places:
+- Installation instructions and editor setup can be found in the
+  [README.md](README.md) and the [Github Pages](https://ocaml.github.io/merlin/)
+- Various topics related to editor support and configuration can be found in the
+  [wiki](https://github.com/ocaml/merlin/wiki).
+- The Merlin protocol is described in [this
+  document](https://github.com/ocaml/merlin/blob/master/doc/dev/PROTOCOL.md).
+
+Contribution to the documentation are welcome!
+
+## Question, bug reports and feature requests
+
+We rely on [Github's issue tracker](https://github.com/ocaml/merlin/issues) for
+support questions, feature requests and bug reports.
+
+When reporting an issue, please include a precise reproduction in the bug report
+when that's possible, as it is a very useful tool to investigate. You should
+also check that you are using the latest version of Merlin and that a similar
+issue as not already been submitted.
+
+## Code contributions
+
+### Styleguides
+
+As of today, Merlin's codebase does not use a code formatter. When contributing
+code to an existing module, one should adopt the style of the surrounding code.
+Please keep lines under 80 characters.
+
+We plan to move the codebase to ocamlformat in a near future.
+
+Changes unrelated to the issue addressed by a PR should be made in a separate
+PR. Additionally, formatting changes in parts of the code not concerned by a
+specific PR should be proposed in another PR.
+
+### Menhir version
+
+Merlin promotes the generated Menhir parser in its sources. This is done to
+avoid depending on Menhir when installing Merlin. However this also means that
+unnecessary diff will appear when the parser gets re-generated by a different
+version of Menhir. To remove this diff please use version `20201216`:
+```bash
+$ opam pin menhir 20201216
+```
+The generated parser file should only be commited if there is an actual change
+in the grammar.
+
+### Repository organization
+
+
+#### `/src/ocaml`
+Vendored OCaml typer. Also include Merlin's incremental parser and other
+utilites.
+
+#### `/src/kernel`
+The backbone of merlin. Contains configuration utilities and pipeline
+management. The pipeline describe the compilation workflow: parsing, ppxing and
+typing.
+
+#### `/src/frontend/query_protocol.ml`
+The type description of Merlin's protocol as documented in
+[PROTOCOL.md](https://github.com/ocaml/merlin/blob/master/doc/dev/PROTOCOL.md)
+
+#### `/src/frontend/query_commands.ml`
+This file contains the dispatcher: a large pattern matching that answers
+Merlin's queries. Most of the time answering a query begins with getting the
+Typedtree from the pipeline and calling the correct analysis functions.
+
+#### `/src/analysis`
+Contains most of Merlin logic required to answer the queries.
+
+#### `/src/dot-protocol`
+The configuration protocol that configuration servers such as
+`dot-merlin-reader` or `dune ocaml-merlin` should implement.
```

Comment:
Is it worth mentioning Dune RPC here as well as an alternative to `dot-merlin-reader` and `dune ocaml-merlin` (it's a genuine question)? We don't have documentation on how Dune communicates with the Merlin server via RPC that we could point to, right?